iroh/src/client.rs

```diff
@@ -578,6 +575,37 @@ impl Stream for BlobAddProgress {
     }
 }
 
+impl Future for BlobAddProgress {
+    type Output = Result<BlobAddOutcome>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        loop {
+            match self.stream.poll_next_unpin(cx) {
+                Poll::Pending => return Poll::Pending,
+                Poll::Ready(None) => {
+                    return Poll::Ready(Err(anyhow!("Response stream ended prematurely")))
+                }
+                Poll::Ready(Some(Err(err))) => return Poll::Ready(Err(err)),
+                Poll::Ready(Some(Ok(msg))) => match msg {
```

Comment:
Misses the `AddProgress::Found` case to add to `self.current_total_size` (see the removed code from `async fn finish`)

Reply:
that's always done now using the atomic u64, see the stream mapping in the constructor